gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java

```diff
@@ -404,7 +411,29 @@ public void delete(URI path, boolean recursive) throws IOException {
               : listFileInfoForPrefixPage(
                       fileInfo.getPath(), DELETE_RENAME_LIST_OPTIONS, /* pageToken= */ null)
                   .getItems();
-      if (!itemsToDelete.isEmpty() && !recursive) {
+
+      /*TODO : making listing of folder and object resources in parallel*/
+      if (isHnBucket) {
+        /**
+         * Get list of folders if the bucket is HN enabled bucket For recursive delete, get all
+         * folder resources. For non-recursive delete, delete the folder directly if it is a
+         * directory and do not do anything if the path points to a bucket.
+         */
+        String bucketName = getBucketName(path);
+        String folderName = getFolderName(path);
+        listOfFolders =
```

Comment:
Let us also add a verbose log to log the number of folders listed.

Reply:
Done